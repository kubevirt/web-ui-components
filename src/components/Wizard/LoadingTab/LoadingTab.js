import React from 'react';

import PropTypes from 'prop-types';

import { Loading } from '../../Loading';

export const LoadingTab = ({ children, text, ...loadingData }) =>
  Object.keys(loadingData).every(dataKey => loadingData[dataKey]) ? children : <Loading key="loading" text={text} />;

LoadingTab.defaultProps = {
  text: 'Loading Page',
  children: null,
};

LoadingTab.propTypes = {
  text: PropTypes.string,
  children: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
};
