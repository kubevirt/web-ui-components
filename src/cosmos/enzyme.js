import createContext from 'react-cosmos-test/enzyme';

import paths from '../../config/paths';

export default function createTestContext(args) {
  args.proxies = args.proxies || []; // workaround of react-cosmos-test bug
  return createContext({
    ...args,
    cosmosConfigPath: `${paths.config}/cosmos.config.js`,
  });
}
