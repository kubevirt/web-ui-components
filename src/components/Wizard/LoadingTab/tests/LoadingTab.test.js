import React from 'react';
import { mount } from 'enzyme';

import { LoadingTab } from '../LoadingTab';

const testLoadingTab = () => <LoadingTab>data</LoadingTab>;

describe('<LoadingTab />', () => {
  it('renders correctly', () => {
    const component = mount(testLoadingTab());
    expect(component).toMatchSnapshot();
  });
});
