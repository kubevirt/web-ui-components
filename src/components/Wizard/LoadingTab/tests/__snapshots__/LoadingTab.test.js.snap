// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingTab /> renders correctly 1`] = `
<LoadingTab
  text="Loading Page"
>
  data
</LoadingTab>
`;
